### DISCUSSION TITLE
JSON modules included in main bundle instead of being bundled with the code that dynamically imports them

### DISCUSSION THREAD

**Discussion by lambdabaa** [webpack/webpack#17326] [Q&A]
# Bug report

I have large JSON files that I am trying to dynamically import in the following way:

```js
// main.js

import('./posts').then(...);
import('./users').then(...);
```

```js
// posts.js
import posts from './datasets/posts';  // This is a JSON module

export default posts;
```

```js
// users.js
import posts from './datasets/users';  // This is a JSON module

export default users;
```

```js
// webpack.config.js

const path = require('path');

module.exports = {
  mode: 'development',
  entry: './main.js',
  output: {
    path: path.resolve(__dirname, '..'),
    filename: '[name].bundle.js',
  },
  module: {
    rules: [{test: /\.js$/, use: 'babel-loader'}],
  },
  watch: true,
};
```

**What is the current behavior?**

`posts.js` and `users.js` get correctly split into their own modules. However, instead of injecting the `./datasets/users.json` into the `users.js` bundle and `./datasets/posts.json` into the `posts.js` bundle, they both end up being thrown into `main.bundle.js`.

**What is the expected behavior?**

`posts.json` should land in the `posts.js` code bundle and `users.json` should be placed in the `users.js` bundle rather than having the main bundle include all of the JSON imports directly.

**Other relevant information:**

  System:
    OS: macOS 13.4
    CPU: (10) arm64 Apple M1 Pro
    Memory: 6.21 GB / 32.00 GB
  Binaries:
    Node: 20.2.0 - /opt/homebrew/bin/node
    npm: 9.6.6 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 114.0.5735.90
    Safari: 16.5
  Packages:
    babel-loader: 9.1.2 => 9.1.2
    webpack: 5.84.1 => 5.84.1
    webpack-cli: ^5.1.1 => 5.1.1

**Comment by alexander-akait** [comment]
What do you mean `This is a JSON module`? Do you want to use them as JSON modules like `import foo from "./package.json" with { type: "json" }` or you want to handle them as assets? 

And please never ignore - how we can reproducible, thank you

**Reply by lambdabaa** [reply]
I meant that `./datasets/posts` maps to a file `./datasets/posts.json` and `./datasets/users` maps to a file `./datasets.users.json`. I want to use them as JSON modules (not handling as assets).

Apologies for not creating a test case.

**Reply by alexander-akait** [reply]
So I don't need any problems with it, you setup splitChunks and extract them for production
